File: Assets/_Resources/Scripts/Enemy/EnemyHealth.js
#pragma strict


public var health : int = 100;
//public var gameObject : GameObject;
function RemoveHealth(amount : int) {
    health = health - amount;
    Debug.Log("Enemy hit");
    if (health <= 0)
    {
        Destroy(this.gameObject);
    }
}

 /*   function OnTriggerEnter (bullet : Collider)    {
        Debug.Log("Hitteddddddddd");
     if (bullet.GetComponent(Collider).tag == "Projectile")  {
         Debug.Log("Hitted");
            RemoveHealth(20);
     }*/
     /*
     function OnCollisionEnter (col : Collision)
         {
             Debug.Log("Hitteddddddddd");
             if(col.gameObject.tag == "Projectile")    {
                 Debug.Log("Hitted");
                 RemoveHealth(20);
             }
             //Destroy(col.gameObject);
         } 
    */
 

File: Assets/_Resources/Scripts/Enemy/EnemyHealth.cs
using UnityEngine;
using System.Collections;

public class EnemyHealth : MonoBehaviour {

    public float health = 100f;

    public void RemoveHealth (float amount)
    {
        health = health - amount;
        if (health <= 0)
        {
            Destroy(gameObject);
        }
    }

}
